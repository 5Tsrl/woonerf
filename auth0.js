'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getLock = exports.defaultLockOptions = exports.authIsRequired = undefined;

var _auth0Lock = require('auth0-lock');

var _auth0Lock2 = _interopRequireDefault(_auth0Lock);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var AUTH0_CLIENT_ID = process.env.AUTH0_CLIENT_ID;
var AUTH0_DOMAIN = process.env.AUTH0_DOMAIN;

var authIsRequired = exports.authIsRequired = AUTH0_CLIENT_ID && AUTH0_DOMAIN;
var defaultLockOptions = exports.defaultLockOptions = {
  auth: {
    params: {
      scope: 'openid analyst offline_access'
    },
    redirect: false
  },
  closeable: false,
  autoclose: true
};
var getLock = exports.getLock = function getLock(lockOptions) {
  return new _auth0Lock2.default(AUTH0_CLIENT_ID, AUTH0_DOMAIN, lockOptions);
};

//# sourceMappingURL=auth0.js