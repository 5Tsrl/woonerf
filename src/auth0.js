import Auth0Lock from 'auth0-lock'
import Auth0Client from 'auth0-js'
import {push} from 'react-router-redux'
import {createAction} from 'redux-actions'

const {AUTH0_CLIENT_ID, AUTH0_DOMAIN} = process.env
const localStorage = window.localStorage

export const setAuth0User = createAction('set auth0 user')
export const authIsRequired = AUTH0_CLIENT_ID && AUTH0_DOMAIN
export const lock = authIsRequired
  ? new Auth0Lock(
    AUTH0_CLIENT_ID,
    AUTH0_DOMAIN, {
      auth: {
        params: {
          scope: 'openid analyst offline_access'
        },
        redirect: false
      },
      closeable: false,
      autoclose: true
    })
  : null
export const client = authIsRequired
  ? new Auth0Client({
    clientID: AUTH0_CLIENT_ID,
    domain: AUTH0_DOMAIN
  })
  : null

/**
 * Use on application mount when authentication is required
 */
export function refreshUser (dispatch) {
  if (authIsRequired) {
    const userString = localStorage.getItem('user')
    const user = userString && JSON.parse(userString)
    if (user && user.refreshToken) {
      dispatch(setAuth0User(user))
      if (process.env.NODE_ENV !== 'development') {
        client.refreshToken(user.refreshToken, function (err, delegationResult) {
          if (err) {
            dispatch(setAuth0User(null))
            localStorage.setItem('user', null)
            dispatch(push('/login'))
          } else {
            user.idToken = delegationResult.id_token
            dispatch(setAuth0User(user))
            localStorage.setItem('user', JSON.stringify(user))
          }
        })
      }
    } else {
      dispatch(push('/login'))
    }
  }
}
