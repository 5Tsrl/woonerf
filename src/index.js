module.exports = {
  actions: require('./actions'),
  auth0: require('./auth0'),
  Auth0Lock: require('./components/auth0-lock'),
  fetch: require('./fetch'),
  html: require('./html'),
  Icon: require('./components/icon'),
  message: require('./message'),
  mount: require('./mount'),
  Pure: require('./components/pure'),
  store: require('./store')
}
