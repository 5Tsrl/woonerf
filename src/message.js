// @flow
import debug from 'debug'
import kebabCase from 'lodash/kebabCase'
import {sprintf} from 'sprintf-js'

const dbg = debug('woonerf:message')

let messages = {}
if (process.env.MESSAGES) {
  setMessages(JSON.parse(process.env.MESSAGES))
}

export default function getMessage (defaultMessage: string, slug: void | string, ...args): string {
  const key = slug || kebabCase(defaultMessage)
  const msg = messages[key] || defaultMessage
  const result = args ? sprintf(msg, ...args) : msg
  dbg(key, result)
  return result
}

export function setMessages (newMessages) {
  messages = newMessages
}
