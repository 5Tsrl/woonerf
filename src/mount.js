import React from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import {browserHistory} from 'react-router'
import {syncHistoryWithStore} from 'react-router-redux'

import createStore from './store'

if (process.env.NODE_ENV === 'development') {
  const Perf = window.Perf = require('react-addons-perf')
  Perf.start()
}

export function create ({
  app,
  reducers
}) {
  const store = createStore(reducers)
  const history = syncHistoryWithStore(browserHistory, store)
  return React.createElement(Provider, {store},
    React.createElement(app, {history, store}))
}

export default function mount ({
  app,
  id = 'root',
  reducers
}) {
  return render(
    create({app, reducers}),
    document.getElementById(id)
  )
}
