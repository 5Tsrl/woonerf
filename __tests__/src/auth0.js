/* globals describe, it */

import {getLock} from '../../src/auth0'

describe('auth0', () => {
  it('getLock should work', () => {
    getLock()
  })
})
