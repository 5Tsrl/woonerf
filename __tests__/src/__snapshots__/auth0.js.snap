exports[`auth0 setAuth0User should work 1`] = `
Object {
  "payload": Object {
    "user": "a user",
  },
  "type": "set auth0 user",
}
`;
